### ISSUE TITLE
SimpleFullscreen not overlaying other apps/windows

### ISSUE THREAD

**Issue by clanktron** [alacritty/alacritty#6367]
### System

OS: macOS Monterey 12.6
Version: alacritty 0.10.1

I'm attempting to replace iterm2 with alacritty but I'm running into issues getting alacritty to overlay with other windows/apps in SimpleFullscreen mode. With iterm2 I can designate a hotkey and have it come into or out of focus on top of whatever my active window is. However whenever I activate alacritty with my hotkey it jumps to my desktop pane instead. My understanding of SimpleFullscreen is that it acts as an isolated pane in front of all others, meaning any time that pane is activated it will just come into focus, rather than move to a particular window.

https://user-images.githubusercontent.com/52805671/192707640-186a1502-b2f1-4d43-b986-8fbeb47eea30.mp4

I'm using hammerspoon for the hotkey functionality.

Hammerspoon Config
```
hs.hotkey.bind({"cmd"}, "e", function()
  alacritty = hs.application.find('alacritty')
  if alacritty ~= nil and alacritty:isFrontmost() then
    alacritty:hide()
  else
    hs.application.launchOrFocus("/Applications/Alacritty.app")
    alacritty.activate(alacritty)
  end
end
)

```

**Comment by chrisduerr** [comment]
I'm not sure I understand what you want. In your video it looks like Alacritty is overlaying other Applications just fine? So seems like it works?

**Comment by clanktron** [comment]
> I'm not sure I understand what you want. In your video it looks like Alacritty is overlaying other Applications just fine? So seems like it works?

It does overlay applications on the desktop pane properly, however when the application is maximized (in this case vscode) it moves back to the desktop pane rather than overlaying the fullscreen application.

**Comment by chrisduerr** [comment]
Sounds like everything is working as intended then? Seems like VSCode is using fullscreen in its own space and thus prevents Alacritty from overlaying it?

**Comment by clanktron** [comment]
No, I'm expecting alacritty to overlay vscode in fullscreen (or any other fullscreen app/window). Below is iterm2 doing what I'm talking about.

https://user-images.githubusercontent.com/52805671/193329677-0d3b312d-7192-428f-adc3-d7fe56697db9.mp4

**Comment by clanktron** [comment]
I don't see why the issue is being closed then, if you aren't interested in personally implementing a fix that's totally fine but leave the issue open so others can see that it needs to be addressed. 

Also if this helps you understand what's wrong: 
SimpleFullScreen is supposed to overlay applications even if they're already in their maximized state, as depicted in my second mp4 with iTerm2. However rather than overlaying those fullscreen applications, Alacritty's SimpleFullscreen seems to limit itself to the desktop pane and will jump back to such whenever it's called into focus.

**Comment by selrond** [comment]
@chrisduerr please, in the spirit of previous comment, reopen this. The issue is apparent.

**Comment by cadebrown** [comment]
+1 I can't use Alacritty as my main driver until this is fixed
